packages/mui-material/src/Badge/Badge.test.js

```diff
@@ -134,7 +134,7 @@ describe('<Badge />', () => {
     it('should render with invisible class when invisible and showZero are set to false and content is 0', () => {
       const { container } = render(<Badge badgeContent={0} showZero={false} invisible={false} />);
       expect(findBadge(container)).to.have.class(classes.invisible);
-      expect(findBadge(container)).to.have.text('');
+      expect(findBadge(container)).to.have.text('0');
```

Comment:
This reverts the fix introduced in https://github.com/mui/material-ui/pull/21557. This fix has added the usage of the `usePreviousProps` hook. You will noticed on this demo: https://next.mui.com/material-ui/react-badge/#badge-visibility, that when clicking on the "-" button, when the value is 1, it immediately disappears without adding the "0" as a content - this is a non valid state. The issue is again here on this PR deploy preview.